java/pom.xml

```diff
@@ -23,7 +23,21 @@
                 <plugin>
                     <groupId>org.openapitools</groupId>
                     <artifactId>openapi-generator-maven-plugin</artifactId>
-                    <version>5.4.0</version>
+                    <version>6.2.1</version>
+                </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-surefire-plugin</artifactId>
+                    <version>${maven-surefire-plugin.version}</version>
+                    <configuration>
+                    </configuration>
+                    <dependencies>
+                        <dependency>
+                            <groupId>org.ow2.asm</groupId>
+                            <artifactId>asm</artifactId>
+                            <version>9.1</version>
```

Comment:
Had to explicitly provide a version of asm that supports Java 17. Otherwise it ends up at 6.2 due to older plexus dragging in old asm version. This is supposedly fixed in newer surefire versions (3+), but Spring uses 2.22.2 still, for some reason..

Reply:
This comment deserves it's place in the code (pom file). Did we conclude what exactly failed without this version? Something with Java Modules, it looked from the stacktrace, right?

Reply:
Made my best effort at the comment. Well, no support for Java 17 in the older version led to asm's `ClassReader` error out. 
Some stack trace for the curious.
```Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 61
    at org.objectweb.asm.ClassReader.<init> (ClassReader.java:176)
    at org.objectweb.asm.ClassReader.<init> (ClassReader.java:158)
    at org.objectweb.asm.ClassReader.<init> (ClassReader.java:146)
    at org.objectweb.asm.ClassReader.<init> (ClassReader.java:273)
    at org.apache.maven.plugin.surefire.booterclient.ModularClasspathForkConfiguration.toModuleName (ModularClasspathForkConfiguration.java:219)
    at org.apache.maven.plugin.surefire.booterclient.ModularClasspathForkConfiguration.createArgsFile (ModularClasspathForkConfiguration.java:158)
    at org.apache.maven.plugin.surefire.booterclient.ModularClasspathForkConfiguration.resolveClasspath (ModularClasspathForkConfiguration.java:98)
    at org.apache.maven.plugin.surefire.booterclient.DefaultForkConfiguration.createCommandLine (DefaultForkConfiguration.java:143)
    at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork (ForkStarter.java:579)
    at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run (ForkStarter.java:282)
    at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run (ForkStarter.java:245)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider (AbstractSurefireMojo.java:1183)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked (AbstractSurefireMojo.java:1011)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute (AbstractSurefireMojo.java:857)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:578)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

Reply:
Made my best effort at the comment. Well, no support for Java 17 in the older version led to asm’s `ClassReader` error out.
Some stack trace for the curious.
```Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 61
    at org.objectweb.asm.ClassReader.<init> (ClassReader.java:176)
    at org.objectweb.asm.ClassReader.<init> (ClassReader.java:158)
    at org.objectweb.asm.ClassReader.<init> (ClassReader.java:146)
    at org.objectweb.asm.ClassReader.<init> (ClassReader.java:273)
    at org.apache.maven.plugin.surefire.booterclient.ModularClasspathForkConfiguration.toModuleName (ModularClasspathForkConfiguration.java:219)
    at org.apache.maven.plugin.surefire.booterclient.ModularClasspathForkConfiguration.createArgsFile (ModularClasspathForkConfiguration.java:158)
    at org.apache.maven.plugin.surefire.booterclient.ModularClasspathForkConfiguration.resolveClasspath (ModularClasspathForkConfiguration.java:98)
    at org.apache.maven.plugin.surefire.booterclient.DefaultForkConfiguration.createCommandLine (DefaultForkConfiguration.java:143)
    at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork (ForkStarter.java:579)
    at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run (ForkStarter.java:282)
    at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run (ForkStarter.java:245)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider (AbstractSurefireMojo.java:1183)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked (AbstractSurefireMojo.java:1011)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute (AbstractSurefireMojo.java:857)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:578)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```